### PR TITLE
Notifications integration

### PR DESCRIPTION
**Pull request by papajulio**

### CODE REVIEW

**Comment by jacalvo** [comment]
code looks :+1: at first glance, but I'm going to do some testing before merging

File: main/mail/src/EBox/Mail.pm [hunk 1]
```diff
@@ -737,12 +737,14 @@ sub _setDovecotConf
     my $gssapiHostname = $sysinfo->hostName() . '.' . $sysinfo->hostDomain();
 
     my $openchange = 0;
+    my $isNotificationsReady = 0;
```
Comment (jacalvo) [inline main/mail/src/EBox/Mail.pm]: maybe $notificationsReady is a better name :) I can do the change when testing it

**Comment by jacalvo** [comment]
Tested successfully, for me it's ready to merge, but maybe we should fix the jenkins problems first. @papajulio will look at it tomorrow.

File: main/openchange/src/EBox/OpenChange.pm [hunk 2]
```diff
@@ -952,8 +955,17 @@ sub certificateIsReserved
     my $managed = defined $vdomains->find(vdomain => $cn);
     return $managed;
 }
-# EBox::CA::Observer methods
```
Comment (sixstone-qq) [inline main/openchange/src/EBox/OpenChange.pm]: This comment should live beneath `notificationsReady` method.

File: main/openchange/src/EBox/OpenChange.pm [hunk 3]
```diff
@@ -952,8 +955,17 @@ sub certificateIsReserved
...
 
+# Method: notificationsReady
+#
+# Returns: If the notifications are ready
+#
+sub notificationsReady
+{
+    my ($self) = @_;
+
+    return (-e OPENCHANGE_DOVECOT_PLUGIN_FILE);
```
Comment (sixstone-qq) [inline main/openchange/src/EBox/OpenChange.pm]: Is `/usr/lib/dovecot/modules/` readable by `ebox` user?

File: main/samba/stubs/openchange.conf.mas [hunk 4]
```diff
@@ -8,7 +9,11 @@
     $openchangeNamedpropsMysqlDb => undef
 </%args>
 [global]
+% if ($openchangeNotificationsReady) {
+    dcerpc endpoint servers = +epmapper, +mapiproxy, +asyncemsmdb
```
Comment (sixstone-qq) [inline main/samba/stubs/openchange.conf.mas]: For me it is more readable in a single line by it is ok as it is.

File: main/openchange/src/EBox/OpenChange.pm (hunk 2 above)
Comment (papajulio) [inline main/openchange/src/EBox/OpenChange.pm]: Added again :)

File: main/openchange/src/EBox/OpenChange.pm (hunk 3 above)
Comment (papajulio) [inline main/openchange/src/EBox/OpenChange.pm]: Yep, it is. At least by our tests.

**Comment by papajulio** [comment]
Jenkins retest this please.

**Pull request merged**